src/test/java/com/ite/itea/ecommerce/usecase/FindAProductUseCaseTest.java

```diff
@@ -0,0 +1,31 @@
+package com.ite.itea.ecommerce.usecase;
+
+import com.ite.itea.ecommerce.domain.retail.ProductId;
+import com.ite.itea.ecommerce.usecase.port.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class FindAProductUseCaseTest {
+
+    @Mock
+    private ProductRepository productRepository;
+
+    private FindAProductUseCase sut;
+
+    @BeforeEach
+    void setUp() {
+        sut = new FindAProductUseCase(productRepository);
+    }
+
+    @Test
+    void shouldGetProductFromRepository() {
+        sut.execute(new ProductId("1"));
+
+        Mockito.verify(productRepository).byId(new ProductId("1"));
```

Comment:
I think we are testing the structure, and the most important behavior (the fact that the use case should return the expected product) is not tested. I suggest setting up the (fake) repository so that it returns the expected product, and then asserting the use case's result rather than what gets called by the use case.

Reply:
damn, totally right!